src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -103,3 +106,18 @@ def evaluate_value(self, value: float | int) -> DataConditionResult:
             return self.get_condition_result()
 
         return None
+
+
+class DetectorWorkflowCondition(abc.ABC):
+    @classmethod
+    @abc.abstractmethod
+    def apply(cls, data: dict[str, Any]) -> Any:
+        pass
+
+    @classmethod
+    @abc.abstractmethod
+    def get_value(data_packet: dict[str, Any], data_condition: DataCondition):
+        pass
+
+
+condition_registry = Registry[Condition]()
```

Comment:
we already have something similar to a registry on data conditions with the `type` field, let's try to only use 1